File: Kattis/Python/catalan_square/catalan_square.py
# https://en.wikipedia.org/wiki/Catalan_number

n = int(input())
c = {0: 1}

for i in range(1, n+1):
    c[i] =  c[i-1] * 2*(2*i+1) // (i+2)  #  The Catalan numbers satisfy the recurrence relations

print(int(c[n]))

File: Kattis/Java/catalanSquare/CatalanSquare.java
//https://open.kattis.com/problems/catalansquare
// https://en.wikipedia.org/wiki/Catalan_number

package catalanSquare;
import java.math.BigInteger;
import kattio.Kattio;

public class CatalanSquare {
	static Kattio io = new Kattio(System.in, System.out);

	public static void main(String[] args) {
		int N = io.getInt();
		BigInteger[] c = new BigInteger[(N == 0 ? 1 : N+1)];
		c[0] = BigInteger.ONE;
		for (int i = 1; i < N +1; i++) { // The Catalan numbers satisfy the recurrence relations
			c[i] = (c[i-1].multiply(BigInteger.valueOf(4*i+2)).divide(BigInteger.valueOf(i+2)));
		}
		System.out.println(c[c.length-1]);
	}
}
